### PR TITLE
Add migration document for 0.109

### PR DESCRIPTION
**Pull request by zachnew-msft**
Adding a folder for documenting migrations due to breaking changes.

### CODE REVIEW

**Comment by mslukebo** [comment]
@zachnew-msft should we add that `CustomDataSourceBase.SetApplicaitonEnvironmentCore` is now virtual, so it can be removed?

**Comment by zachnew-msft** [comment]
> @zachnew-msft should we add that `CustomDataSourceBase.SetApplicaitonEnvironmentCore` is now virtual, so it can be removed?

That's not a breaking change, so I didn't think of it. I can add a note if it would be helpful.

**Comment by mslukebo** [comment]
> > @zachnew-msft should we add that `CustomDataSourceBase.SetApplicaitonEnvironmentCore` is now virtual, so it can be removed?
> 
> That's not a breaking change, so I didn't think of it. I can add a note if it would be helpful.

It might be useful to have a [Breaking Changes] section and a [Optional Changes] section in this. If people are going to be making code changes when upgrading they have one place to look for everything they might want/need to change.

**Comment by zachnew-msft** [comment]
> > > @zachnew-msft should we add that `CustomDataSourceBase.SetApplicaitonEnvironmentCore` is now virtual, so it can be removed?
> > 
> > 
> > That's not a breaking change, so I didn't think of it. I can add a note if it would be helpful.
> 
> It might be useful to have a [Breaking Changes] section and a [Suggested Changes] section in this. If people are going to be making code changes when upgrading they have one place to look for everything they might want/need to change.

That's a great idea. Thanks!

File: documentation/Migration/Update-0.109.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+# Suggested Changes
+
+The following are changes that are not required, but may be useful to you.
+
+## CustomDataSourceBase
+
+The method `SetApplicationEnvironmentCore` is now virtual, so you no longer have
+to override it if you do not want to. An `ApplicationEnvironment` property is
+now available on the base class that you may reference in your `CustomDataSources.`
```
Comment (mslukebo) [inline documentation/Migration/Update-0.109.md:97]: nit: `CustomDataSources.` -> `CustomDataSource`.

File: documentation/Migration/Update-0.109.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}
+catch (UnsupportedDataSourceException)
+{
+    // ...
+}
+````
+
+## IDataSource
+
+Anywhere you are calling `IDataSource.GetUri()` can be simply updated to `IDataSource.Uri`
```
Comment (mslukebo) [inline documentation/Migration/Update-0.109.md:87]: I would prefer these be kept solely as lists of changes, at least for each intro sentence. This should start with "`IDataSource.GetUri()` has been changed to a read-only property: `IDataSource.Uri`.

File: documentation/Migration/Update-0.109.md (hunk 1 above)
Comment (zachnew-msft) [reply documentation/Migration/Update-0.109.md:97]: Thanks!

File: documentation/Migration/Update-0.109.md (hunk 2 above)
Comment (zachnew-msft) [reply documentation/Migration/Update-0.109.md:87]: I'll update it

**Review by jmaxson-ms — DISMISSED** [review]

**Review by jmaxson-ms — APPROVED** [review]

**Review by trgibeau — APPROVED** [review]

**Pull request merged**